### ISSUE TITLE
Portfolio view is also showing deleted projects

### ISSUE THREAD

**Issue by danielwiese** [Puzzlepart/prosjektportalen365#363]
**Describe the bug**
As a member of the group "Portføjeinnsyn" you will be able to see all the projects in the portfolio view. 
The bug in this instance is that you see deleted projects as well, meaning deleted sites is still visible.

**To Reproduce**

1. Go to https://prosjektportalen.sharepoint.com/sites/pp365/SitePages/Portef%C3%B8ljeoversikt.aspx as prosjektportalen admin user.
2. Look at the project overview list and find e.g "Bitpro demoprosjekt"
3. Go to https://prosjektportalen-admin.sharepoint.com/_layouts/15/online/AdminHome.aspx#/recycleBin as prosjektportalen admin user
4. Find "Bitpro demoprosjekt" in the deleted sites section
5. EXPERIENCE, VITNESS & LIVE THE BUG!

**Expected behavior**
I would like deleted project not to be visible in the portfolio view

**Screenshots**

**About your setup**

- Installed version [found under {Prosjektportalen}/Lists/Installasjonslogg/AllItems.aspx]

**Additional context**

**Comment by tarjeieo** [comment]
Good catch, indeed a consequence we did not consider. Not really sure how to avoid this, as we cannot check for the site's existence as the current user does not necessarily have access. Perhaps opening up the hidden Projects-list for admins is necessary.

**Comment by olemp** [comment]
> Good catch, indeed a consequence we did not consider. Not really sure how to avoid this, as we cannot check for the site's existence as the current user does not necessarily have access. Perhaps opening up the hidden Projects-list for admins is necessary.

@tarjeieo Maybe open up the project list for admins somehow so the entry can be deleted? Can't think of any other option tbh.

**Comment by danielwiese** [comment]
@tarjeieo & @olemp , any updates on this issue, or is there an alternative way to delete a project to avoid the issue untill there is a better solution?

**Comment by tarjeieo** [comment]
Temp fix is to delete the project from the Lists/Prosjekter list

**Comment by tarjeieo** [comment]
How about having a symbol next  to the projects in the portfolio overview? Since handling both deletion of the project information and the site itself is quite complex.

Feature:
* Only visible for admins
* Icon: X (cross)
* Tooltip: "Delete project"
* Confirmation message: "Deletion of a project is done in two steps. First, delete the project from the project itself, by going to the project site, opening the settings cog-wheel menu, select "Site information" and finally select "Delete site". Once this is done, you can remove the project from this list by selecting the button below."
* Button in the confirmation dialog: "Yes, remove references to this project".

**Comment by olemp** [comment]
> How about having a symbol next to the projects in the portfolio overview? Since handling both deletion of the project information and the site itself is quite complex.
> 
> Feature available as a button from the portfolio overview:
> 
> * Only visible for admins
> * Icon: X (cross)
> * Tooltip: "Delete project"
> * Confirmation message: "Deletion of a project is done in two steps. First, delete the project from the project itself, by going to the project site, opening the settings cog-wheel menu, select "Site information" and finally select "Delete site". Once this is done, you can remove the project from this list by selecting the button below."
> * Button in the confirmation dialog: "Yes, remove references to this project".

Shouldn't it be on the project? In the project information web part.

**Comment by tarjeieo** [comment]
Yeah, maybe that's better, but if the project site is deleted there's no available option to remove the project from the projects list.

**Comment by tarjeieo** [comment]
Unless we're able to delete both the site and the project item simultaneously.

**Comment by olemp** [comment]
> Unless we're able to delete both the site and the project item simultaneously.

That was my thought.

**Comment by olemp** [comment]
> How about having a symbol next to the projects in the portfolio overview? Since handling both deletion of the project information and the site itself is quite complex.
> 
> Feature available as a button from the portfolio overview:
> 
> * Only visible for admins
> * Icon: X (cross)
> * Tooltip: "Delete project"
> * Confirmation message: "Deletion of a project is done in two steps. First, delete the project from the project itself, by going to the project site, opening the settings cog-wheel menu, select "Site information" and finally select "Delete site". Once this is done, you can remove the project from this list by selecting the button below."
> * Button in the confirmation dialog: "Yes, remove references to this project".

Added this solution to the issue description. Deletion from the project site is too complex.

**Comment by okms** [comment]
We should have one single way of deleting Prosjektportalen 365 projects. This can be from the Projects list.

**Comment by olemp** [comment]
> We should have one single way of deleting Prosjektportalen 365 projects. This can be from the Projects list.

You mean implementing some kind of "hook" that also deleted the site collection and the resources?
